src/interactableComponents/drawer/index.js

```diff
@@ -0,0 +1,475 @@
+            onPress={onLeftPress}
+            underlayColor={Colors.getColorTint(background, 50)}
+          >
+            <View
+              style={{
+                height: this.state.height,
+                width: leftItemWidth,
+                padding: ITEM_PADDING,
+                justifyContent: 'center',
+                alignItems: 'center',
+              }} 
+            >
+              {leftItem && leftItem.icon &&
+              <Animated.Image
+                source={leftItem.icon}
+                style={
+                [{
+                  tintColor: this.props.itemsTintColor,
+                  width: this.props.itemsIconSize,
+                  height: this.props.itemsIconSize,
+                  opacity: this.deltaX.interpolate({
+                    inputRange: [leftItemWidth - BLEED, leftItemWidth],
+                    outputRange: [0, 1],
+                    extrapolateLeft: 'clamp',
+                    extrapolateRight: 'clamp',
+                  }),
+                  transform: [{
+                    scale: this.deltaX.interpolate({
+                      inputRange: [leftItemWidth - BLEED, leftItemWidth],
+                      outputRange: [0.7, 1],
+                      extrapolateLeft: 'clamp',
+                      extrapolateRight: 'clamp',
+                    }),
+                  }],
+                },
+                ]}
+              />}
+              {leftItem && leftItem.text && 
+              <Animated.Text
+                numberOfLines={1}
+                style={
+                [{
+                  color: this.props.itemsTintColor,
+                  ...this.state.typography,
+                  ...this.props.itemsTextStyle,
+                  marginTop: this.state.textTopMargin,
+                  opacity: this.deltaX.interpolate({
+                    inputRange: [leftItemWidth - BLEED, leftItemWidth],
+                    outputRange: [0, 1],
+                    extrapolateLeft: 'clamp',
+                    extrapolateRight: 'clamp',
+                  }),
+                  transform: [{
+                    scale: this.deltaX.interpolate({
+                      inputRange: [leftItemWidth - BLEED, leftItemWidth],
+                      outputRange: [0.7, 1],
+                      extrapolateLeft: 'clamp',
+                      extrapolateRight: 'clamp',
+                    }),
+                  }],
+                },
+                ]}
+              >
+                {leftItem.text}
+              </Animated.Text>}
+            </View>
+          </TouchableHighlight>
+        </Animated.View>
+      </View>
+    );
+  }
+  renderRightItem(item, index) {
+    const inputRanges = this.getInputRanges();
+
+    return (
+      <TouchableOpacity
+        key={index}
+        style={[
+          this.styles.item, {
+            width: this.getItemWidth(item),
+            backgroundColor: item.background || ITEM_BG,
+            padding: this.state.itemPadding,
```

Comment:
Destruct `state` & `props` at the beginning of the method like we usually do, it helps with readability.